packages/integration/src/lib/search/search.state.ts

```diff
@@ -129,7 +129,9 @@ export class SearchState {
           fw instanceof FeatureWorkspace &&
           fw.layer.options.workspace.searchIndexEnabled
       );
-      this.searchSourceService.setWorkspaces(wksSource, searchableWks);
+      if (wksSource) {
```

Comment:
Au final, si le wksSource est undefined, est-ce qu'on devrait sauter toute le restant de la méthode?
```suggestion

if (wksSource) {
  return;
}
this.workspaceState.store.entities$.subscribe((e) => {
  const searchableWks = e.filter(
    (fw) =>
      fw instanceof FeatureWorkspace &&
      fw.layer.options.workspace.searchIndexEnabled
  );
  this.searchSourceService.setWorkspaces(wksSource, searchableWks);
});
this.monitorLayerDeletion();
```

Reply:
fait.